CMakeLists.txt

```diff
@@ -41,12 +45,16 @@ ENDIF(WIN32)
 
 # --- OPTIONS ----------------------------------------
 OPTION (BUILD_BENCHMARK "Build the benchmarks" OFF)
-OPTION (BUILD_UNIT_TESTS "Build the unitary tests" OFF)
 OPTION (BUILD_UTILS "Build the utils" OFF)
 OPTION (BUILD_PYTHON_INTERFACE "Build the python binding" ON)
 OPTION (BUILD_WITH_LUA_SUPPORT "Build the lua parser" OFF)
 OPTION (BUILD_WITH_COMMIT_VERSION "Build libraries by setting specific commit version" OFF)
 
+IF(DEFINED BUILD_UNIT_TESTS)
```

Comment:
We do not need to set it deprecated:
- this won't impact the behavior of Pinocchio
- it will output a anoying messsage for non-dev people

Reply:
I think a short deprecation notice is useful.
It will make it easier to handle the change for tools that automatically build Pinocchio.
Also, everyone's build folders are configured with `BUILD_UNIT_TESTS`. With no deprecation period, this setting will just stop working and everybody will have fall on `BUILD_TESTING` set to on, with no clue on why. At least this will output an explicit message. Maybe it should be even _more_ explicit and prescriptive, such as "we suggest that you delete you build folder and make a new one".
I have tested this to ensure that the change will be as smooth as possible.
The message is completely invisible to anybody making a new build folder from scratch.
For the deprecation, I took inspiration from `DISABLE_TESTS`: https://github.com/jrl-umi3218/jrl-cmakemodules/blob/54ece258eed16da94c7f10979588fcb47b744eb8/test.cmake#L35

Reply:
OK. Fine for the explanations.